controller/src/beerocks/master/tasks/optimal_path_task.cpp

```diff
@@ -227,8 +227,88 @@ void optimal_path_task::work()
                 TASK_LOG(INFO) << "Remove candidate " << it->first << " , vap " << candidate_bssid
                                << " is not in steer list: " << database.config.load_steer_on_vaps;
                 it = potential_11k_aps.erase(it);
+            } else {
+                ++it;
             }
-            ++it;
+        }
+
```

Comment:
realy tryied to make it smaller , 
but because i need sometimes to  break or end the function  or even continue its not that simple .
I added a comment in the code that optimal path refactoring task should address it 
https://jira.prplfoundation.org/browse/PPM-102